hrrecruitment/CRM/HRRecruitment/Upgrader.php

```diff
@@ -116,4 +116,47 @@ public function upgrade_1400() {
     CRM_Core_DAO::executeQuery("UPDATE civicrm_case_type SET weight = 7 WHERE name = 'Application'");
     return TRUE;
   }
+
+  /**
+   * Remove the "Application" assignment type and the managed entity.
+   *
+   * @return bool
+   */
+  public function upgrade_1402() {
+    // remove the managed entity entry
+    $query = 'DELETE FROM civicrm_managed ' .
+      'WHERE module = %1 AND name = %2 AND entity_type = %3';
+    $params = [
+      1 => ['org.civicrm.hrrecruitment', 'String'],
+      2 => ['Application', 'String'],
+      3 => ['CaseType', 'String']
+    ];
+    CRM_Core_DAO_Managed::executeQuery($query, $params);
+
+    $result = civicrm_api3('CaseType', 'get', [
+      'name' => 'Application',
+    ]);
+
+    // Application type doesn't exist, so our job is done
+    if ($result['count'] < 1) {
```

Comment:
I was thinking about this, but it seems like assignments are deleted when you delete the parent CaseType. The code after this to fetch the assignments won't work if type "Application" CaseType doesn't exist.